src/Azure.DataApiBuilder.Mcp/BuiltInTools/CreateRecordTool.cs

```diff
@@ -37,51 +53,218 @@ public Tool GetToolMetadata()
             };
         }
 
-        public Task<CallToolResult> ExecuteAsync(
+        public async Task<CallToolResult> ExecuteAsync(
             JsonDocument? arguments,
             IServiceProvider serviceProvider,
             CancellationToken cancellationToken = default)
         {
+            ILogger<CreateRecordTool>? logger = serviceProvider.GetService<ILogger<CreateRecordTool>>();
             if (arguments == null)
             {
-                return Task.FromResult(new CallToolResult
-                {
-                    Content = [new TextContentBlock { Type = "text", Text = "Error: No arguments provided" }]
-                });
+                return BuildErrorResult("Invalid Arguments", "No arguments provided", logger);
             }
 
             try
             {
-                // Extract arguments
+                cancellationToken.ThrowIfCancellationRequested();
                 JsonElement root = arguments.RootElement;
 
                 if (!root.TryGetProperty("entity", out JsonElement entityElement) ||
                     !root.TryGetProperty("data", out JsonElement dataElement))
                 {
-                    return Task.FromResult(new CallToolResult
-                    {
-                        Content = [new TextContentBlock { Type = "text", Text = "Error: Missing required arguments 'entity' or 'data'" }]
-                    });
+                    return BuildErrorResult("Invalid Arguments", "Missing required arguments 'entity' or 'data'", logger);
                 }
 
                 string entityName = entityElement.GetString() ?? string.Empty;
+                if (string.IsNullOrEmpty(entityName))
```

Comment:
I think it woud be better to check if it is whitespace as well
```suggestion
                if (string.IsNullOrWhitespace(entityName))
```